### DISCUSSION TITLE
Migration: Changing Column Type and then applying a per row update

### DISCUSSION THREAD

**Discussion by Leylan24** [simolus3/drift#2758] [Q&A] [answered]
I am trying to perform a column migration where it migrates the column from an int -> string. After applying the migration I run a for loop to manually assign a string UUID value to each row. Is there a way for me to use the columnTransformer property and apply row updates?

**Comment by simolus3** [comment, marked as answer]
Do you need to know the UUID, for instance to patch other tables having a reference to that column? If not, you could register a user-defined function and call that in `columnTransformer`.

To register the function, add a `setup` parameter to `NativeDatabase`, like this:

```dart
setup: (rawDb) {
  final uuid = Uuid();

  rawDb.createFunction(
      functionName: 'new_uuid', function: (args) => uuid.v4());
},
```

Then, you can use `FunctionCallExpression('new_uuid', [])` in the column transformer to use random UUIDs during the migration.

**Reply by Leylan24** [reply]
So just to clarify. We currently have an existing database that is as follows:

| id | key |
| -------- | ------- |
| 1| value1 |
| 2| value2  |
| 3| value3  |

after the coloumnTransformer is applied using the method you provided above the final table will look as follows:

| id | key |
| -------- | ------- |
| 113da8e1-6ba1-4720-916c-a285860fcb74 | value1 |
| ddef295c-e24c-489c-996f-7dce61f73ac6 | value2  |
|  3307ef97-9b6f-4a2b-ae3d-4f5bec8bb886 | value3  |
